### ISSUE TITLE
"--watch" options does not works.

### ISSUE THREAD

**Issue by hg-pyun** [babel/babel#8789]
## Bug Report
**Current Behavior**
Setting up --watch options with a Babel and modifying a file should automatically be compiled. But now I don't think I can automatically compile.

**Input Code**
- REPL or Repo link if applicable
  - https://github.com/hg-pyun/iterize/blob/c31f4f8ad4430323afb677b61a6b8e1b13bd6379/package.json#L26
  - https://github.com/hg-pyun/iterize/tree/c31f4f8ad4430323afb677b61a6b8e1b13bd6379

**Expected behavior/code**
Automatically compiles when the file that Babel is watching is modified.

**Babel Configuration (.babelrc, package.json, cli command)**
.babelrc
```js
{
  "presets": [
    "@babel/env",
    "@babel/preset-typescript"
  ],
  "plugins": [
    ["@babel/plugin-transform-modules-umd", {
      "globals": {
        "iterize": "Iterize"
      },
      "exactGlobals": true
    }]
  ],
  "env": {
    "production": {
      "presets": [
        "minify"
      ]
    }
  }
}
```
package.json
```js
{
  ...
  "devDependencies": {
    "@babel/cli": "^7.1.2",
    "@babel/core": "^7.1.2",
    "@babel/plugin-transform-modules-umd": "^7.1.0",
    "@babel/preset-env": "^7.1.0",
    "@babel/preset-typescript": "^7.1.0",
    "babel-preset-minify": "^0.5.0",
    "cross-env": "^5.2.0"
  },
  "scripts": {
    "dev": "babel ./src/index.ts --watch -o ./lib/iterize.dev.js"
  }
}

```
**Environment**
- Babel version(s): 7.1.2
- Node/npm version: 8.11.4
- OS: OSX 10.13.6
- How you are using Babel: cli

**Comment by idahogurl** [comment]
I am having the same issue.

**package.json**
```json
{
  "name": "scripts",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "cheerio": "^1.0.0-rc.2",
    "good": "^8.1.2",
    "he": "^1.2.0",
    "moment": "^2.24.0",
    "request": "^2.88.0",
    "request-promise-native": "^1.0.7"
  },
  "devDependencies": {
    "@babel/cli": "^7.2.3",
    "@babel/core": "^7.3.4",
    "@babel/preset-env": "^7.3.4"
  },
  "scripts": {
    "build": "yarn babel ./index.js --out-file index.compiled.js",
    "dev": "nodemon index.compiled.js"
  }
}
```

**.babelrc**
```json
{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "node": "current"
        }
      }
    ]
  ]
}
```

**Comment by suginoki45** [comment]
Hi! Thank you for the great tool!
I also report the same issue as it occurs.

**Environment**
- macOS 10.14 Mojave
- Node v10.14.2
- npm v6.9.0

**package.json**
```
{
  "name": "scripts",
  "version": "1.0.0",
  "description": "",
  "main": "app.js",
  "config": {
    "paths": "./file/to/path"
  },
  "scripts": {
    "watch:babel": "babel $npm_package_config_paths/src/js/bundle.js --watch --out-dir $npm_package_config_paths/dist/js"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4"
  }
}
```

**.babelrc**
```
{
  "presets": ["@babel/preset-env"]
}
```

**Comment by alexisab** [comment]
Same problem :
```
{
    "name": "mypackage",
    "version": "1.0.0",
    "description": "",
    "main": "lib/index.js",
    "private": false,
    "files": [
        "/lib"
    ],
    "scripts": {
        "watch": "yarn transpile --watch",
        "transpile": "babel src -d lib",
        "clean": "rm -rf ./lib",
        "lint": "eslint src",
        "test": "mocha src/index.test.js"
    },
    "dependencies": {
        "core-js": "^3.1.4",
        "invariant": "^2.2.4",
        "regenerator-runtime": "^0.13.2"
    },
    "devDependencies": {
        "@babel/cli": "^7.4.4",
        "@babel/core": "^7.4.5",
        "@babel/preset-env": "^7.4.5",
        "chai": "^4.2.0",
        "eslint": "^6.0.1",
        "mocha": "^6.1.4"
    },
    "peerDependencies": {
        "koa": "^2.7.0",
        "pino": "^5.12.6"
    },
    "babel": {
        "presets": [
            "@babel/preset-env"
        ]
    }
}
```

`yarn transpile` works , but `yarn watch` doesn't transpile src on updates

**Comment by hg-pyun** [comment]
@hzoo Please, check this issue :D

**Comment by loganfsmyth** [comment]
Are folks here running these scripts in Docker or in a virtual machine or anything like that?

**Comment by hg-pyun** [comment]
In my case, running on `OSX 10.13.6`

**Comment by alexisab** [comment]
In fact it seems to work bu it doesn't log anything in the console to say that the code have been re-compiled

**Comment by eprikazc** [comment]
Does not work for me, in Ubuntu 18.04. Does not log anything in the console, and does not recompile file.

**Comment by jasonmcaffee** [comment]
I encountered this issue with typescript.  Babel will watch js files by default, but not .ts files.
To change this, provide the --extensions option to babel, along with the watch:
```
"build-watch": "babel src --out-dir build --source-maps --watch --extensions '.ts,.js'",
```
I'm using versions:
```
"devDependencies": {
    "@babel/cli": "^7.6.2",
    "@babel/core": "^7.6.2",
    "@babel/preset-typescript": "^7.6.0",
...
}
```
Hope that helps!

**Comment by JLHwung** [comment]
@jasonmcaffee Thank you for the report. The issue you reported will be fixed by #10270.

**Comment by JLHwung** [comment]
@hg-pyun Thanks for the reproduction repo. The issue you mentioned has been fixed in #10283.

If you hit here from search engines, please open a new issue with reproducing repo. Thanks!
